core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -80,7 +56,57 @@ public boolean isSampled() {
      *
      * @param rate minimum sample rate is 0.01, or 1% of traces
      */
-    public static Sampler create(float rate) {
+    static Sampler<Object> random(double rate) {
         return CountingSampler.create(rate);
     }
+
+    /**
+     * Returns a sampler, given a rate-limited on a per-second interval.
+     *
+     * @param samplesPerSecond minimum rate-limited is 0 and the max is 2,147,483,647 (max int)
+     */
+    static Sampler<Object> rateLimited(int samplesPerSecond) {
```

Comment:
I realized that this could be more useful to users if we did:

```java
static <T> Sampler<T> rateLimited(double rate) {
    @SuppressWarnings(...)
    final Sampler<T> cast = (Sampler<T>) RateLimitingSampler.create(rate);
    return cast;
}
```